front/components/assistant/conversation/messages/MessageGroup.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useEffect, useRef } from "react";
+
+import type { MessageWithContentFragmentsType } from "@app/components/assistant/conversation/ConversationViewer";
+
+interface MessageGroupProps {
+  messages: MessageWithContentFragmentsType[][];
+  isLastMessage: boolean;
+  children?: React.ReactNode;
+}
+
+const OFFSET_FROM_WINDOW = 450;
+
+export default function MessageGroup({
+  messages,
+  isLastMessage,
+  children,
+}: MessageGroupProps) {
+  const lastMessageRef = useRef<HTMLDivElement>(null);
+
+  const shouldMaximize = React.useMemo(() => {
+    if (!messages || !messages[0] || !messages[0][0]) {
+      return true;
+    }
+    const firstMessage = messages[0][0];
+
+    if (firstMessage.type === "user_message") {
+      return false;
+    }
+
+    const isGenerating = firstMessage.status === "created";
+    const isLastMessageAgentMessage = firstMessage.type === "agent_message";
+
+    return isLastMessage && isLastMessageAgentMessage && isGenerating;
+  }, [messages, isLastMessage]);
+
+  const dynamicMinHeight = shouldMaximize
+    ? `${window.innerHeight - OFFSET_FROM_WINDOW}px`
+    : "0px";
+
+  useEffect(() => {
+    if (isLastMessage && lastMessageRef.current) {
+      lastMessageRef.current.scrollIntoView({ behavior: "smooth" });
+    }
+  }, [isLastMessage]);
+
+  return (
+    <div
+      className={isLastMessage ? "last-message-group" : ""}
+      ref={isLastMessage ? lastMessageRef : undefined}
+      style={{ minHeight: dynamicMinHeight }}
```

Comment:
Should we use tailwind?

Reply:
I presume the value of style vs tailwind is that you can mutate it from the observer?

Wouldn't it be cleaner to have the observer return a value that we can then use here?

Reply:
I initially tried using Tailwind classes here (something like `min-h-[200px]`) but couldn't have it work

Reply:
Not sure I fully understood but seems like Tailwind can't recognize and apply styles for dynamically generated class names that depend on JavaScript runtime values.